pkg/plugin/conntrack/conntrack_linux.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// package conntrack implements a conntrack plugin for Retina.
+package conntrack
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/rlimit"
+	"github.com/microsoft/retina/internal/ktime"
+	"github.com/microsoft/retina/pkg/log"
+	plugincommon "github.com/microsoft/retina/pkg/plugin/common"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go@master -cc clang-14 -cflags "-g -O2 -Wall -D__TARGET_ARCH_${GOARCH} -Wall" -target ${GOARCH} -type ct_key conntrack ./_cprog/conntrack.c -- -I../lib/_${GOARCH} -I../lib/common/libbpf/_src
+
+var (
+	ct   *Conntrack
+	once sync.Once
+)
+
+type Conntrack struct {
+	l     *log.ZapLogger
+	objs  *conntrackObjects
+	ctmap *ebpf.Map
+}
+
+func Init() (*Conntrack, error) {
+	once.Do(func() {
+		ct = &Conntrack{}
+	})
+	if ct.l == nil {
+		ct.l = log.Logger().Named("conntrack")
+	}
+	if ct.objs != nil {
+		return ct, nil
+	}
+
+	// Allow the current process to lock memory for eBPF resources.
+	if err := rlimit.RemoveMemlock(); err != nil {
+		ct.l.Error("RemoveMemlock failed", zap.Error(err))
+		return ct, errors.Wrapf(err, "failed to remove memlock limit")
+	}
+
+	objs := &conntrackObjects{}
+	err := loadConntrackObjects(objs, &ebpf.CollectionOptions{
+		Maps: ebpf.MapOptions{
+			PinPath: plugincommon.ConntrackMapPath,
+		},
+	})
+	if err != nil {
+		ct.l.Error("loadConntrackObjects failed", zap.Error(err))
+		return ct, err
+	}
+
+	ct.objs = objs
+
+	// Get the conntrack map from the objects
+	ct.ctmap = objs.conntrackMaps.RetinaConntrackMap
+
+	return ct, nil
+}
+
+// Close cleans up the Conntrack plugin.
+func (ct *Conntrack) Close() {
+	if ct.objs != nil {
+		ct.objs.Close()
+	}
+}
+
+// gc loops through the conntrack map and deletes entries older than the specified timeout.
+func (ct *Conntrack) gc(timeout time.Duration) {
+	ct.l.Debug("Running Conntrack GC loop")
+
+	var key conntrackCtKey
+	var value conntrackCtValue
+
+	var noOfCtEntries, noOfCtEntriesDeleted int
+
+	iter := ct.ctmap.Iterate()
```

Comment:
this will be a nightmare to iterate through all the entries. We should use speciality Maps, which can help do this